graveler/value.go

```diff
@@ -0,0 +1,88 @@
+package graveler
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+/*
+ * Value is serialized in a trivial fixed-order format:
+ *
+ *    | len(Identity) | Identity | len(Value) | Value |
+ *
+ * where each length is serialized as a varint and additional bytes after Value are silently
+ * ignored.
```

Comment:
1. From the `binary` package docs, it seems that Varint is basically the Protobuf implementation of a Varint. This means any Graveler implementation (other than the Go one) will either depend on Protobuf to parse these, or will have to implement the Protobuf implementation of a Varint.

2. Why silently ignore? Why not just assume everything after `Identity` is the value?

Reply:
1. `Varint` is used also in Google protobufs.  I feel safe using it, given that the implementation is entirely self-contained and is [just a few short routines](https://golang.org/src/encoding/binary/varint.go) that bang bits together.  Would you feel safer if I used Uvarint, which is even shorter?
2. I insist on encoding `Value` in order to get extensibility.  Otherwise in future I will not be able to add some third field needed by graveler.  "But why not encode that putative future extra field inside `Value`, and have graveler return the inner `Value.ActualOldValue` field?" Well, then for backwards compatibility (new code reading old files) I would need either file-level metadata (yuck!) or to rely on some known property of the old values as actually specified by the code calling graveler.  And it still wouldn't work for old code reading new files (e.g. a dump routine that prints `"[key] -> [value]\n"`.  Doing it this way provides compatibility the protobuf way: it is always safe to add optional ignorable fields.